src/Interpreters/Context.h

```diff
@@ -688,6 +693,10 @@ class Context: public std::enable_shared_from_this<Context>
     // Reload Zookeeper
     void reloadZooKeeperIfChanged(const ConfigurationPtr & config) const;
 
+#if USE_HIVE
+    HMSClientPtr getHMSClient(const String & name) const;
+#endif
```

Comment:
Can we find a way not to put it in context?
May be just a singleton getter class instead? It will be even better I think.

Reply:
Hive metastore client is similar to zookeeper client.  Since we can get zookeeper client from Context with functions `getZooKeeper` and `getAuxiliaryZooKeeper`,  I think it is also ok to get hive metastore client from Context.

Reply:
No, we shouldn't put anything to context unless there is important reason.
So singleton class will be better for `HiveMetastoreClient` (pls rename `HMSClientPtr` to `HiveMetastoreClientPtr`).
btw someone told me that `getZooKeeper` is not really needed in context too and can be better put in other place.